src/main/java/maksym/fedorenko/bookstore/controller/BookController.java

```diff
@@ -35,13 +37,15 @@ public BookDto getBookById(@PathVariable Long id) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
-    @PutMapping("/{id}")
+    @PatchMapping("/{id}")
     @ResponseStatus(HttpStatus.ACCEPTED)
-    public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
+    public BookDto updateBook(
+            @PathVariable Long id, @RequestBody @Valid UpdateBookRequestDto bookDto
```

Comment:
you may also validate such things as @PathVariable, @RequestParam etc.

it will look like: `@PathVariable @Positive Long id`

however in order to do so you need to annotate your controller with @Validated annotation

this is just FYI :)

you decide - to apply it or not